src/test/java/com/softserve/academy/spaced/repetition/controller/CourseControllerTest.java

```diff
@@ -0,0 +1,286 @@
+package com.softserve.academy.spaced.repetition.controller;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.softserve.academy.spaced.repetition.domain.*;
+import com.softserve.academy.spaced.repetition.service.CourseService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CourseControllerTest {
+    private MockMvc mockMvc;
+
+    @InjectMocks
+    private CourseController courseController;
+
+    @Mock
+    private CourseService courseService;
+
+    @Before
+    public void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(courseController)
+                .setControllerAdvice(new ExceptionHandlerController())
+                .alwaysDo(print())
+                .build();
+    }
+
+    @Test
+    public void getCourseById() throws Exception {
+        long categoryId = 1L;
+        long courseId = 1L;
+        when(courseService.getCourseById(eq(categoryId),eq(courseId))).thenReturn(createCourse());
+        mockMvc.perform(get("/api/category/{category_id}/courses/{course_id}", 1, 1)
```

Comment:
I think, you had better used your variables categoryId and courseId instead of just 1.